x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/PlanTimeProfile.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.operator;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Profile information for plan optimization phases.
+ * Captures timing information for logical and physical optimization steps.
+ *
+ */
+public final class PlanTimeProfile implements Writeable, ToXContentObject {
+    private long planNanos;
+    private long logicalOptimizationNanos;
+    private long physicalOptimizationNanos;
+
+    /**
+     * @param logicalOptimizationNanos  Time spent on local logical plan optimization (in nanoseconds)
+     * @param physicalOptimizationNanos Time spent on local physical plan optimization (in nanoseconds)
```

Comment:
I would like to move my prior question here: https://github.com/elastic/elasticsearch/pull/138400#discussion_r2564401910

Reply:
> Do you mind clarifying planNanos as well?
> 
> It looks like it includes:
> 
> breakPlanBetweenCoordinatorAndDataNode
> reductionPlan
> plan.collect(LookupJoinExec.class::isInstance)...
> All of them should be fairly cheap.

I'm looking at it now - as you say, it's fairly cheap and probably makes no sense as it addresses multiple intermediate operations for planning.

I will remove it as it adds some confusion. Thanks!